ccmbioinfo/lfs
# CARDS-2042: Patient portal: add support for recording communication events related to patient surveys

Test on the **CARDS-2042-testing** branch!

To test:

- create and fill in patient information form with a valid email and consent to receive emails
- create a visit information form for a clinic set 3 days ago
- check that a Survey events form was created, with the right information
    - Hospital
    - Department
    - Discharged on
    - Assigned survey
    - Survey expires on
- change the visit date to 7 days ago, save, wait 1 minute
- check that the `Invitation email sent on` question  now has an answer
- change the visit date to 9 days ago, save, wait 1 minute
- check that the `First reminder email sent on` question  now has an answer
- change the visit date to 11 days ago, save, wait 1 minute
- check that the `Second reminder email sent on` question  now has an answer
- change the visit date to 60 days ago, save, wait 1 minute
- check that the data form (e.g. CPESIC) was deleted, but the Survey events form is still present

- generate a new visit information survey
- generate a token and fill in and submit the survey as a patient
- check that the `Patient responses received on` question now has an answer
